### PR TITLE
add reticulate-based tests for the hdf5 backend

### PR DESCRIPTION
**Pull request by rcannood**
In discussing with @LouiseDck yesterday, I'm thinking that the only way to properly make sure that anndataR is able to read files created by anndata and vice versa, is to use reticulate to implement tests with data going back and forth.

Even without the extra checks, seeing if the python anndata is able to read files produced by anndataR is already very valuable.

These tests won't work yet since some of the functions it calls don't exist yet

### CODE REVIEW

**Comment by lazappi** [comment]
I think you're right this is the only way to test stuff but I'm not entirely sure doing it as a normal test will work in practice. I'm not sure you can guarantee that Python/**anndata** will be available without using something like **{basilisk}**. It may be we need a separate CI or something that does this round trip (I think @ivirshup was interested in that anyway). Happy to have this for now though if it works.

**Comment by rcannood** [comment]
It should be just a matter of making sure that `anndata::install_anndata()` or `reticulate::install_miniconda(); reticulate::py_install("anndata", pip = TRUE)` is called in the CI to make this work.

Happy to give basilisk a shot as well :)

**Comment by rcannood** [comment]
If the tests succeed, this test could be merged to the `hdf5-structure` branch and then extended

**Comment by rcannood** [comment]
@lazappi Ok to merge?

**Review by lazappi — APPROVED** [review]

File: tests/testthat/test-HDF5AnnData-with-reticulate.R [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  geneinfo = sample(c("a", "b", "c"), 20, replace = TRUE)
+)
+
+# construct dummy var names
+var_names <- paste0("gene", seq_len(20))
+
+layers <- list(
+  X2 = X * 2,
+  X3 = X * 3
+)
```
[the remark this answers is not in the document]
Comment (rcannood) [reply tests/testthat/test-HDF5AnnData-with-reticulate.R]: Completely agree :)

**Pull request merged**